### PR TITLE
Add 'Operators on minikube' doc and update other docs

### PR DESCRIPTION
**Pull request by dharmit**
**What type of PR is this?**
/kind documentation

**What does this PR do / why we need it**:
Mainly adds "Operators on minikube" document and other documents are updated by `./script/sync.sh`. Nothing has been done manually in this PR.

**Which issue(s) this PR fixes**:

Fixes: NA

**PR acceptance criteria**:

- [ ] Unit test 

- [ ] Integration test 

- [x] Documentation 

- [ ] I have read the [test guidelines](https://github.com/openshift/odo/blob/master/docs/dev/test-architecture.adoc)

**How to test changes / Special notes to the reviewer**:
If anything other than `chosen-docs/upstream/operators-on-minikube.md` looks out of place, please open an issue on the repo because everything else has been generated by script.

### CODE REVIEW

File: docs/creating-a-single-component-application-with-odo.md [hunk 1]
```diff
@@ -122,7 +120,7 @@ You can modify your application code and have the changes applied to your applic
 
 # Adding storage to the application components
 
-Persistent storage keeps data available between restarts of odo. Use the `odo storage` command to add persistent data to your application. Examples of data that must persist include database files, dependencies, and build artifacts, such as a `.m2` Maven directory.
+Use the `odo storage` command to add persistent data to your application. Examples of data that must persist include database files, dependencies, and build artifacts, such as a `.m2` Maven directory.
```
Comment (girishramnani) [inline docs/creating-a-single-component-application-with-odo.md:123]: `Examples of data that is usually persisted`?

Comment (girishramnani) [reply docs/creating-a-single-component-application-with-odo.md:123]: maybe explain what does it mean when you say `persistent`, like is that when a developer wants to keep their data safe between `odo push`?

Comment (dharmit) [reply docs/creating-a-single-component-application-with-odo.md:123]: The way this works is we have to fix this in the `adoc` file in the master branch. This PR (or any PR on `gh-pages` branch for that matter) is just a script that converted these adoc to md. So it updates the existing md files if there's any change in master branch. Or, like in this PR, we add new doc.

To address your concern, all we can do at the moment is open a new issue in the repo and fix it. We need to keep doc review out of the scope of PR review for gh-pages. :disappointed:

**Review by mik-dass — APPROVED** [review]
/lgtm

**Comment by dharmit** [comment]
@kadel @mik-dass @adisky can one you try to approve this now that #4258 is merged?

**Pull request merged**